src/dlgPackageExporter.cpp

```diff
@@ -268,64 +317,230 @@ void dlgPackageExporter::slot_export_package()
+            // defined finite-sized char type buffer {which obviously can
+            // have string buffer overflow issues} is available in 1.x
+            // versions of libzip):
+            zip_error_t error;
+            zip_error_init_with_code(&error, ze);
+            displayResultMessage(tr("Failed to open module (archive) file, error is: \"%1\".")
+                                 .arg(QString::fromUtf8(zip_error_strerror(&error))), false);
+            zip_error_fini(&error);
+            isOk = false;
+            // The above flag will now cause execution to drop down to the bottom of
+            // the method and set up a means to close the dialogue after the user
+            // has seen the error message...
+
+        } else {
+            // Opened/created archive file successfully
+/*
+ * Previous code here failed if the user included sub-directories in the
+ * temporary directory where the contents for the package/module was assembled -
+ * as sub-directories are now correctly handled by the installer code we needed
+ * to revise the code here to also work when the user uses them. NB This is
+ * typically the case where they have collections of images or sound files
+ * which they wish to store in a heirachical manner...! - Slysven
+ */
+#if defined(Q_OS_WIN32)
+/*
+ * From Qt Docs:
+ * Note: On NTFS file systems, ownership and permissions checking is disabled by
+ * default for performance reasons. To enable it, include the following line:
+ */
+            extern Q_CORE_EXPORT int qt_ntfs_permission_lookup;
+/*
+ * Permission checking is then turned on and off by incrementing and
+ * decrementing qt_ntfs_permission_lookup by 1:
+ */
+            qt_ntfs_permission_lookup++;
+#endif // defined(Q_OS_WIN32)
+            QDirIterator itDir(mStagingDirName, QDir::NoDotAndDotDot|QDir::AllDirs|QDir::Files, QDirIterator::Subdirectories);
+            // relative names to use in archive:
+            QStringList directoryEntries;
+            // Key is relative name to use in archive
+            // Value is fullName in file-system:
+            QMap<QString, QString> fileEntries;
+            while (itDir.hasNext() && isOk) {
+                QString itEntry = itDir.next();
+                Q_UNUSED(itEntry);
+//              Comment out the preceding line if the following is uncommented!
+//              qDebug() << " parsing entry:" << itEntry << " fileName() is:" << itDir.fileName() << " filePath() is:" << itDir.filePath();
+                if (!(  itDir.fileName().compare(QStringLiteral("."))
+                     && itDir.fileName().compare(QStringLiteral("..")))) {
+
+                     // Dot and DotDot entries are no use to us so skip them
+                     continue;
+                }
+
+                QFileInfo entryInfo(itDir.fileInfo());
+                if (! entryInfo.isReadable()) {
+                    qWarning() << "dlgPackageExporter::slot_export_package() skipping file: "
```

Comment:
Make warning texts translatable, please?

Reply:
These are not shown to the usual user - they're only shown if you launch Mudlet from the terminal. Let's keep it English so we can understand what users show us.

Reply:
Fair enough.